test/test_qdmi.cpp

```diff
@@ -25,16 +25,27 @@ SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 #include <complex>
 #include <cstddef>
 #include <cstdint>
+#include <functional>
 #include <gmock/gmock.h>
 #include <gtest/gtest.h>
 #include <random>
 #include <sstream>
 #include <string>
 #include <tuple>
 #include <unordered_map>
+#include <unordered_set>
 #include <utility>
 #include <vector>
 
+/// Hash function for a pair
+template <class T, class U> struct std::hash<std::pair<T, U>> {
+  auto operator()(const std::pair<T, U> &p) const noexcept -> std::size_t {
+    const std::size_t h1 = std::hash<T>{}(p.first);
+    const std::size_t h2 = std::hash<U>{}(p.second);
+    return h1 ^ h2;
+  }
+};
```

Comment:
To avoid the manipulation of the global namespace here, you can make use of the third template argument of the `unordered_map` class (https://en.cppreference.com/w/cpp/container/unordered_map/unordered_map.html), which is the function being used for hashing.